Generate serialization of SecurityOriginData again

achristensen07 opened the pull request:
#### 6a91176c19ddd21a298d2ba6adb64fbf427a6953
<pre>
Generate serialization of SecurityOriginData again
<a href="https://bugs.webkit.org/show_bug.cgi?id=252284">https://bugs.webkit.org/show_bug.cgi?id=252284</a>

Reviewed by Chris Dumez.

In 260214@main Matt made a SecurityOriginData either a tuple of protocol/host/port
or an identifier of its opaque origin.  To serialize the data in a clean way,
we need to make its data a variant of the tuple or the opaque origin.

Changing the hash function changed the order of hash tables which changed the layout
test results of storage/indexeddb/getdatabases.html like it did in 250093@main

* Source/WTF/wtf/CrossThreadCopier.h:
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Markable.h:
* Source/WebCore/inspector/agents/InspectorDatabaseAgent.cpp:
(WebCore::InspectorDatabaseAgent::didOpenDatabase):
* Source/WebCore/loader/cache/MemoryCache.cpp:
(WebCore::MemoryCache::removeResourcesWithOrigin):
* Source/WebCore/page/ClientOrigin.h:
(WebCore::ClientOrigin::clientRegistrableDomain const):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAllowNavigationToCustomProtocolWithoutUserGesture):
* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::initializeShared):
(WebCore::SecurityOrigin::isSameOriginDomain const):
(WebCore::SecurityOrigin::canDisplay const):
(WebCore::SecurityOrigin::isSameOriginAs const):
(WebCore::SecurityOrigin::isPotentiallyTrustworthy const):
(WebCore::SecurityOrigin::domainForCachePartition const):
(WebCore::SecurityOrigin::toString const):
(WebCore::SecurityOrigin::create):
(WebCore::SecurityOrigin::equal const):
* Source/WebCore/page/SecurityOrigin.h:
(WebCore::SecurityOrigin::protocol const):
(WebCore::SecurityOrigin::host const):
(WebCore::SecurityOrigin::port const):
(WebCore::SecurityOrigin::isHTTPFamily const):
* Source/WebCore/page/SecurityOriginData.cpp:
(WebCore::SecurityOriginData::toString const):
(WebCore::SecurityOriginData::databaseIdentifier const):
(WebCore::SecurityOriginData::isolatedCopy const):
(WebCore::SecurityOriginData::isolatedCopy):
(WebCore::operator==):
* Source/WebCore/page/SecurityOriginData.h:
(WebCore::SecurityOriginData::Tuple::operator== const):
(WebCore::SecurityOriginData::Tuple::isolatedCopy const):
(WebCore::SecurityOriginData::SecurityOriginData):
(WebCore::SecurityOriginData::createOpaque):
(WebCore::SecurityOriginData::protocol const):
(WebCore::SecurityOriginData::host const):
(WebCore::SecurityOriginData::port const):
(WebCore::SecurityOriginData::isNull const):
(WebCore::SecurityOriginData::isOpaque const):
(WebCore::SecurityOriginData::isHashTableDeletedValue const):
(WebCore::add):
(WebCore::SecurityOriginData::encode const): Deleted.
(WebCore::SecurityOriginData::decode): Deleted.
* Source/WebCore/platform/RegistrableDomain.h:
(WebCore::RegistrableDomain::RegistrableDomain):
(WebCore::RegistrableDomain::matches const):
* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder&lt;WebCore::SecurityOriginData&gt;::encode):
* Source/WebCore/platform/network/CredentialStorage.cpp:
(WebCore::CredentialStorage::removeCredentialsWithOrigin):
* Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm:
(WebCore::NetworkStorageSession::deleteCookies):
* Source/WebCore/platform/network/mac/CredentialStorageMac.mm:
(WebCore::CredentialStorage::removeSessionCredentialsWithOrigins):
* Source/WebCore/workers/service/ServiceWorkerRegistrationKey.cpp:
(WebCore::ServiceWorkerRegistrationKey::firstPartyForCookies const):
(WebCore::ServiceWorkerRegistrationKey::toDatabaseKey const):
* Source/WebCore/workers/service/server/SWServer.cpp:
(WebCore::SWServer::addRegistration):
(WebCore::SWServer::removeRegistration):
(WebCore::SWServer::removeFromScopeToRegistrationMap):
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::deleteWebsiteDataForOrigin):
(WebKit::NetworkProcess::deleteWebsiteDataForOrigins):
(WebKit::filterForRegistrableDomains):
(WebKit::NetworkProcess::deleteAndRestrictWebsiteDataForRegistrableDomains):
(WebKit::NetworkProcess::registrableDomainsWithWebsiteData):
* Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerConnection.cpp:
(WebKit::WebSharedWorkerServerConnection::requestSharedWorker):
* Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerToContextConnection.cpp:
(WebKit::WebSharedWorkerServerToContextConnection::launchSharedWorker):
* Source/WebKit/NetworkProcess/cache/CacheStorageEngineCaches.cpp:
(WebKit::CacheStorage::Caches::storeOrigin):
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::createWebSocketTask):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::deleteDataForRegistrableDomains):
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::void&gt;::encode):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
(headers_for_type):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/API/c/WKSecurityOriginRef.cpp:
(WKSecurityOriginCopyProtocol):
(WKSecurityOriginCopyHost):
(WKSecurityOriginGetPort):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/C/WKFrameInfoRef.cpp:
(WKFrameInfoCopySecurityOrigin):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageUIClient):
* Source/WebKit/UIProcess/API/Cocoa/WKFrameInfo.mm:
(-[WKFrameInfo securityOrigin]):
* Source/WebKit/UIProcess/API/Cocoa/WKSecurityOrigin.mm:
(-[WKSecurityOrigin protocol]):
(-[WKSecurityOrigin host]):
(-[WKSecurityOrigin port]):
* Source/WebKit/UIProcess/API/Cocoa/_WKFrameTreeNode.mm:
(-[_WKFrameTreeNode securityOrigin]):
* Source/WebKit/UIProcess/Cocoa/MediaPermissionUtilities.mm:
(WebKit::applicationVisibleNameFromOrigin):
(WebKit::alertMessageText):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::requestStorageSpace):
* Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp:
(WebKit::WebAuthenticatorCoordinatorProxy::handleRequest):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataRecord.cpp:
(WebKit::WebsiteDataRecord::displayNameForOrigin):
(WebKit::WebsiteDataRecord::matches const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::fetchDataAndApply):

Canonical link: <a href="https://commits.webkit.org/260458@main">https://commits.webkit.org/260458@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3656c2430d6fb9584dec5bb9756118a32097b58c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108364 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17459 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41227 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117480 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116849 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112251 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/18960 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8738 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100587 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114131 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14150 "Build was cancelled. Recent messages:") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97403 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42128 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/111894 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/18960 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29048 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/97537 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10291 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30392 "Passed tests") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/38/builds/96877 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/8401 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11033 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7294 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/38/builds/96877 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16441 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49989 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/105953 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7234 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12620 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26224 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/SecurityOrigin.cpp:
```diff
@@ -553,8 +559,8 @@ Ref<SecurityOrigin> SecurityOrigin::create(const String& protocol, const String&
     auto origin = create(URL { protocol + "://" + host + "/" });
-    if (port && !WTF::isDefaultPortForProtocol(*port, protocol))
-        origin->m_data.port = port;
+    if (port && !WTF::isDefaultPortForProtocol(*port, protocol) && std::holds_alternative<SecurityOriginData::Tuple>(origin->m_data.data))
```
This new code looks a lot more complicated, for the same functionality. This is not a good compromise.
While using a variant is a good idea in theory, the execution here is not quite right in my opinion.

SecurityOrigin shouldn't need to know anything about the internals of SecurityOriginData, this is bad encapsulation. It used to be OK because SecurityOriginData was a simple struct. However, now that it uses a variant internally and has getters, I don't think it is still OK for SecurityOrigin to go poke inside SecurityOriginData's internals like this.

cdumez commented on Source/WebCore/page/SecurityOrigin.cpp:
```diff
@@ -396,8 +399,11 @@ bool SecurityOrigin::isSameOriginAs(const SecurityOrigin& other) const
 
-    if (isOpaque() || other.isOpaque())
-        return data().opaqueOriginIdentifier == other.data().opaqueOriginIdentifier;
+    if (m_data.data.index() != other.m_data.data.index())
```
This is another example where the new code is much more complicated and shouldn't need to be.

cdumez commented on Source/WebCore/page/SecurityOriginData.cpp:
```diff
@@ -39,15 +39,15 @@ namespace WebCore {
 String SecurityOriginData::toString() const
 {
-    if (protocol == "file"_s)
+    if (protocol() == "file"_s)
```
Given that these getters are not necessarily super cheap, we may want to consider caching their result in variables to avoid calling them repeatedly in this function.

cdumez commented on Source/WebCore/page/SecurityOriginData.cpp:
```diff
@@ -101,10 +101,10 @@ String SecurityOriginData::databaseIdentifier() const
     // Now that we've fixed that bug, we produce this string for compatibility
     // with existing persistent state.
-    if (equalLettersIgnoringASCIICase(protocol, "file"_s))
+    if (equalLettersIgnoringASCIICase(protocol(), "file"_s))
```
ditto for protocol.

cdumez commented on Source/WebCore/page/SecurityOriginData.h:
```diff
@@ -36,49 +36,88 @@ namespace WebCore {
+        String host;
+        std::optional<uint16_t> port;
+        bool operator==(const Tuple& other) const { return protocol == other.protocol && host == other.host && port == other.port; }
+        Tuple isolatedCopy() const { return { protocol.isolatedCopy(), host.isolatedCopy(), port }; }
```
We should have an isolatedCopy() version that works on a r-value reference so that it can be leverage by the one we have on SecurityOriginData.

cdumez commented on Source/WebCore/page/SecurityOriginData.h:
```diff
@@ -36,49 +36,88 @@ namespace WebCore {
+enum OpaqueOriginIdentifierType { };
+using OpaqueOriginIdentifier = ObjectIdentifier<OpaqueOriginIdentifierType>;
+
 struct SecurityOriginData {
```
Can this be a class now?

cdumez commented on Source/WebCore/page/SecurityOriginData.h:
```diff
@@ -110,52 +157,19 @@ struct SecurityOriginData {
 inline void add(Hasher& hasher, const SecurityOriginData& data)
 {
-    add(hasher, data.protocol, data.host, data.port, data.opaqueOriginIdentifier);
+    add(hasher, data.data.index());
```
we already have a `add(Hasher&)` overload for `std::variant` so let's not duplicate the logic here.
We should add a `add(Hasher&)` overload for `SecurityOriginData::tuple` and then just call `add(hasher, data)` here.

cdumez commented on Source/WebCore/page/SecurityOriginData.h:
```diff
@@ -36,49 +36,88 @@ namespace WebCore {
-    String host;
-    std::optional<uint16_t> port;
-    MarkableOpaqueOriginIdentifier opaqueOriginIdentifier;
+    std::variant<Tuple, ProcessQualified<OpaqueOriginIdentifier>> data;
```
Can this be private and be named `m_data` now? I don't think outsiders should be interacting with this variant directly anymore.

cdumez reviewed (approved):
I still have a few improvement ideas but this is definitely much nicer than the previous iteration IMO.

achristensen07 committed:
Generate serialization of SecurityOriginData again
https://bugs.webkit.org/show_bug.cgi?id=252284

Reviewed by Chris Dumez.

In 260214@main Matt made a SecurityOriginData either a tuple of protocol/host/port
or an identifier of its opaque origin.  To serialize the data in a clean way,
we need to make its data a variant of the tuple or the opaque origin.

Changing the hash function changed the order of hash tables which changed the layout
test results of storage/indexeddb/getdatabases.html like it did in 250093@main

* Source/WTF/wtf/CrossThreadCopier.h:
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Markable.h:
* Source/WebCore/inspector/agents/InspectorDatabaseAgent.cpp:
(WebCore::InspectorDatabaseAgent::didOpenDatabase):
* Source/WebCore/loader/cache/MemoryCache.cpp:
(WebCore::MemoryCache::removeResourcesWithOrigin):
* Source/WebCore/page/ClientOrigin.h:
(WebCore::ClientOrigin::clientRegistrableDomain const):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAllowNavigationToCustomProtocolWithoutUserGesture):
* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::initializeShared):
(WebCore::SecurityOrigin::isSameOriginDomain const):
(WebCore::SecurityOrigin::canDisplay const):
(WebCore::SecurityOrigin::isSameOriginAs const):
(WebCore::SecurityOrigin::isPotentiallyTrustworthy const):
(WebCore::SecurityOrigin::domainForCachePartition const):
(WebCore::SecurityOrigin::toString const):
(WebCore::SecurityOrigin::create):
(WebCore::SecurityOrigin::equal const):
* Source/WebCore/page/SecurityOrigin.h:
(WebCore::SecurityOrigin::protocol const):
(WebCore::SecurityOrigin::host const):
(WebCore::SecurityOrigin::port const):
(WebCore::SecurityOrigin::isHTTPFamily const):
* Source/WebCore/page/SecurityOriginData.cpp:
(WebCore::SecurityOriginData::toString const):
(WebCore::SecurityOriginData::databaseIdentifier const):
(WebCore::SecurityOriginData::isolatedCopy const):
(WebCore::SecurityOriginData::isolatedCopy):
(WebCore::operator==):
* Source/WebCore/page/SecurityOriginData.h:
(WebCore::SecurityOriginData::Tuple::operator== const):
(WebCore::SecurityOriginData::Tuple::isolatedCopy const):
(WebCore::SecurityOriginData::SecurityOriginData):
(WebCore::SecurityOriginData::createOpaque):
(WebCore::SecurityOriginData::protocol const):
(WebCore::SecurityOriginData::host const):
(WebCore::SecurityOriginData::port const):
(WebCore::SecurityOriginData::isNull const):
(WebCore::SecurityOriginData::isOpaque const):
(WebCore::SecurityOriginData::isHashTableDeletedValue const):
(WebCore::add):
(WebCore::SecurityOriginData::encode const): Deleted.
(WebCore::SecurityOriginData::decode): Deleted.
* Source/WebCore/platform/RegistrableDomain.h:
(WebCore::RegistrableDomain::RegistrableDomain):
(WebCore::RegistrableDomain::matches const):
* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder<WebCore::SecurityOriginData>::encode):
* Source/WebCore/platform/network/CredentialStorage.cpp:
(WebCore::CredentialStorage::removeCredentialsWithOrigin):
* Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm:
(WebCore::NetworkStorageSession::deleteCookies):
* Source/WebCore/platform/network/mac/CredentialStorageMac.mm:
(WebCore::CredentialStorage::removeSessionCredentialsWithOrigins):
* Source/WebCore/workers/service/ServiceWorkerRegistrationKey.cpp:
(WebCore::ServiceWorkerRegistrationKey::firstPartyForCookies const):
(WebCore::ServiceWorkerRegistrationKey::toDatabaseKey const):
* Source/WebCore/workers/service/server/SWServer.cpp:
(WebCore::SWServer::addRegistration):
(WebCore::SWServer::removeRegistration):
(WebCore::SWServer::removeFromScopeToRegistrationMap):
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::deleteWebsiteDataForOrigin):
(WebKit::NetworkProcess::deleteWebsiteDataForOrigins):
(WebKit::filterForRegistrableDomains):
(WebKit::NetworkProcess::deleteAndRestrictWebsiteDataForRegistrableDomains):
(WebKit::NetworkProcess::registrableDomainsWithWebsiteData):
* Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerConnection.cpp:
(WebKit::WebSharedWorkerServerConnection::requestSharedWorker):
* Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerToContextConnection.cpp:
(WebKit::WebSharedWorkerServerToContextConnection::launchSharedWorker):
* Source/WebKit/NetworkProcess/cache/CacheStorageEngineCaches.cpp:
(WebKit::CacheStorage::Caches::storeOrigin):
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::createWebSocketTask):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::deleteDataForRegistrableDomains):
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::void>::encode):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
(headers_for_type):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/API/c/WKSecurityOriginRef.cpp:
(WKSecurityOriginCopyProtocol):
(WKSecurityOriginCopyHost):
(WKSecurityOriginGetPort):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/C/WKFrameInfoRef.cpp:
(WKFrameInfoCopySecurityOrigin):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageUIClient):
* Source/WebKit/UIProcess/API/Cocoa/WKFrameInfo.mm:
(-[WKFrameInfo securityOrigin]):
* Source/WebKit/UIProcess/API/Cocoa/WKSecurityOrigin.mm:
(-[WKSecurityOrigin protocol]):
(-[WKSecurityOrigin host]):
(-[WKSecurityOrigin port]):
* Source/WebKit/UIProcess/API/Cocoa/_WKFrameTreeNode.mm:
(-[_WKFrameTreeNode securityOrigin]):
* Source/WebKit/UIProcess/Cocoa/MediaPermissionUtilities.mm:
(WebKit::applicationVisibleNameFromOrigin):
(WebKit::alertMessageText):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::requestStorageSpace):
* Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp:
(WebKit::WebAuthenticatorCoordinatorProxy::handleRequest):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataRecord.cpp:
(WebKit::WebsiteDataRecord::displayNameForOrigin):
(WebKit::WebsiteDataRecord::matches const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::fetchDataAndApply):

Canonical link: https://commits.webkit.org/260458@main
